src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -305,6 +308,7 @@ public override async Task<ModelSimulation> MapToModel(SnapshotSimulation snapsh
          simulation.AddAnalyses(await populationAnalysesFrom(simulation, snapshot.PopulationAnalyses, snapshotContext));
 
          _simulationParameterOriginIdUpdater.UpdateSimulationId(simulation);
+         _chartTask.UpdateObservedDataInChartsFor(simulation, snapshotContext.Project);
```

Comment:
I am not sure WHY this is required now and not before. I am assuming that a change in core did not add the repository automatically. Maybe @georgeDaskalakis Can comment on this? At any rate, this should be done anyways IMO so this is not a bug per see in core

Reply:
@msevestre I am not quite sure....maybe we needed this because of the grouping of the chart lines  - let's briefly discuss this in the meeting today.